test/cpp/end2end/xds/xds_override_host_end2end_test.cc

```diff
@@ -126,52 +117,143 @@ class OverrideHostTest : public XdsEnd2endTest {
   CheckRpcSendOk(DEBUG_LOCATION, 5, RpcOptions().set_metadata(session_cookie));
   EXPECT_EQ(backends_[backend_idx]->backend_service()->request_count(), 5);
   // Round-robin spreads the load
   ResetBackendCounters();
-  CheckRpcSendOk(DEBUG_LOCATION, backends_.size() * 2);
+  CheckRpcSendOk(DEBUG_LOCATION, 4);
   EXPECT_EQ(2, backends_[0]->backend_service()->request_count());
   EXPECT_EQ(2, backends_[1]->backend_service()->request_count());
+  // Should be disabled for now
+  EXPECT_EQ(0, backends_[2]->backend_service()->request_count());
   // Call a different service with the same cookie
   ResetBackendCounters();
   CheckRpcSendOk(DEBUG_LOCATION, 5,
                  RpcOptions()
                      .set_metadata(session_cookie)
                      .set_rpc_service(RpcService::SERVICE_ECHO2));
   EXPECT_EQ(backends_[backend_idx]->backend_service2()->request_count(), 5);
+  // DRAINING is not overridden here so this channel should just disappear
+  std::array<HealthStatus, 3> statuses(
+      {HealthStatus::HEALTHY, HealthStatus::HEALTHY, HealthStatus::HEALTHY});
+  statuses[backend_idx] = HealthStatus::DRAINING;
+  balancer_->ads_service()->SetEdsResource(
+      EdsResourceWithChannelHealth(statuses));
+  WaitForAllBackends(DEBUG_LOCATION);
+  // Draining channel is not overridden
+  CheckRpcSendOk(DEBUG_LOCATION, (backends_.size() - 1) * 2,
+                 RpcOptions().set_metadata(session_cookie));
+  EXPECT_EQ(0, backends_[backend_idx]->backend_service()->request_count());
+  // The second of the "initial backends"
+  EXPECT_EQ(2, backends_[1 - backend_idx]->backend_service()->request_count());
+  EXPECT_EQ(2, backends_[2]->backend_service()->request_count());
+}
+
+TEST_P(OverrideHostTest, DefaultConfigurationDrainingOverridden) {
+  CreateAndStartBackends(3);
+  Cluster cluster = default_cluster_;
+  auto* lb_config = cluster.mutable_common_lb_config();
+  auto* override_health_status_set = lb_config->mutable_override_host_status();
+  override_health_status_set->add_statuses(HealthStatus::HEALTHY);
+  override_health_status_set->add_statuses(HealthStatus::UNKNOWN);
+  override_health_status_set->add_statuses(HealthStatus::DRAINING);
+  balancer_->ads_service()->SetCdsResource(cluster);
+  SetListenerAndRouteConfiguration(balancer_.get(),
+                                   BuildListenerWithStatefulSessionFilter(),
+                                   default_route_config_);
+  balancer_->ads_service()->SetEdsResource(EdsResourceWithChannelHealth(
+      {HealthStatus::HEALTHY, HealthStatus::HEALTHY}));
+  WaitForAllBackends(DEBUG_LOCATION, 0, 2);
+  // First call gets the cookie. RR policy picks the backend we will use.
+  auto server_initial_metadata =
+      SendRpcGetServerMetadata(DEBUG_LOCATION, RpcOptions());
+  auto session_cookie = GetHeadersWithSessionCookie(server_initial_metadata);
+  ASSERT_FALSE(session_cookie.empty());
+  size_t backend_idx = FindBackendWithRequest();
```

Comment:
Consider forcing the test to get a cookie for one particular backend, so that the code below can use fixed expectations for each backend:

```
// Get a cookie for backend 0.
std::vector<std::pair<std::string, std::string>> session_cookie;
do {
  auto server_initial_metadata =
      SendRpcGetServerMetadata(DEBUG_LOCATION, RpcOptions());
  session_cookie = GetHeadersWithSessionCookie(server_initial_metadata);
  ASSERT_FALSE(session_cookie.empty());
} while (FindBackendWithRequest() != 0);
```